java/arcs/android/host/ArcHostHelper.kt

```diff
@@ -0,0 +1,210 @@
+                    it as ParticleIdentifier
+                    arrayList += it.toParcelable()
+                }
+                bundle.putParcelableArrayList(OPERATION_RESULT, arrayList)
+            }
+            is Parcelable -> bundle.putParcelable(OPERATION_RESULT, result)
+            is Serializable -> bundle.putSerializable(OPERATION_RESULT, result)
+            else -> {
+                // Unit
+            }
+        }
+        // This triggers a suspended coroutine to resume with the value.
+        intent.getParcelableExtra<ResultReceiver>(OPERATION_RECEIVER)?.send(0, bundle)
+    }
+
+    enum class Operation {
+        START_ARC, STOP_ARC, REGISTER_PARTICLE, UNREGISTER_PARTICLE,
+        GET_REGISTERED_PARTICLES
+    }
+
+    companion object {
+        const val ARC_HOST_INTENT = "arcs.android.host.ARC_HOST"
+        const val OPERATION = "OPERATION"
+        const val OPERATION_ARG = "OPERATION_ARG"
+        const val OPERATION_RECEIVER = "OPERATION_RECEIVER"
+        const val OPERATION_RESULT = "OPERATION_RESULT"
+
+        fun createArcHostIntent(
+            operation: Operation,
+            component: ComponentName,
+            argument: Parcelable?
+        ): Intent = Intent(ARC_HOST_INTENT).setComponent(component)
+            .putExtra(OPERATION, operation)
+            .putExtra(
+                OPERATION_ARG,
+                argument
+            )
+
+        fun onResult(intent: Intent, receiver: ResultReceiver?) =
+            intent.putExtra(OPERATION_RECEIVER, receiver)
+
+        fun createStartArcHostIntent(
+            service: ComponentName,
+            planPartition: PlanPartition
+        ): Intent = createArcHostIntent(
+            Operation.START_ARC,
+            service,
+            planPartition.toParcelable()
+        )
+
+        fun createStopArcHostIntent(
+            service: ComponentName,
+            planPartition: PlanPartition
+        ): Intent = createArcHostIntent(
+            Operation.STOP_ARC,
+            service,
+            planPartition.toParcelable()
+        )
+
+        fun createRegisterParticleIntent(
+            componentName: ComponentName,
+            particleIdentifier: ParticleIdentifier
+        ) = createArcHostIntent(
+            Operation.REGISTER_PARTICLE, componentName, particleIdentifier.toParcelable()
+        )
+
+        fun createUnregisterParticleIntent(
+            componentName: ComponentName,
+            particleIdentifier: ParticleIdentifier
+        ) = createArcHostIntent(
+            Operation.UNREGISTER_PARTICLE, componentName, particleIdentifier.toParcelable()
+        )
+
+        fun createGetRegisteredParticlesIntent(componentName: ComponentName): Intent =
+            createArcHostIntent(
+                Operation.GET_REGISTERED_PARTICLES,
+                componentName,
+                null
+            )
+    }
+}
+
+fun KClass<out Service>.toComponentName(context: Context) = ComponentName(context, this.java)
```

Comment:
I think I saw a few places where this could be leveraged, but isn't yet.  Also, all public functions should have KDoc.